floodlight/core/events.py

```diff
@@ -61,3 +74,155 @@ def custom(self):
+        """
+        check = True
+        for col in essential_events_columns:
+            if col not in self.essential:
+                check = False
+                warnings.warn(
+                    f"Events Object is missing the essential event column {col}!"
+                )
+        return check
+
+    def check_essential_cols(self) -> bool:
+        """Perform the checks against the definitions (from floodlight.core.definitions)
+        for all essential columns in the inner events DataFrame
+
+        Returns
+        -------
+        check: bool
+            True if the checks for all essential columns pass and False otherwise
+        """
+        check = True
+        for col in self.essential:
+            check *= self.definition_check(col, essential_events_columns)
+
+        return check
+
+    def check_protected_cols(self) -> bool:
+        """Perform the checks against the definitions (from
+        floodlight.core.definitions) for all protected columns in the inner events
+        DataFrame
+
+        Returns
+        -------
+        check: bool
+            True if the checks for all protected columns pass and False otherwise
+        """
+        check = True
+        for col in self.protected:
+            check *= self.definition_check(col, protected_columns)
+
+        return check
+
+    def definition_check(self, col: str, definitions: Dict[str, Dict]) -> bool:
+        """Perform the checks for value range for a single column of the inner event
+        DataFrame using the specifications from floodlight.core.definitions.
+
+        Parameters
+        ----------
+        col: str
+            Column of the inner event DataFrame to be checked
+        definitions: Dict
+            Dictionary for each column name to be checked. Each entry is a Dictionary
+            containing definitions for allowed value ranges of the respective
+            column. Information about value_range is given as a List of the form
+            [min_value, max_value, extra_value] where the extra_value indicates an
+            additionally allowed value that does not fit within the defined value range.
+
+        Returns
+        -------
+        check: bool
+            True if the checks for value range pass and False otherwise
+        """
+        check = True
+
+        # check if value range is defined
+        if definitions[col]["value_range"] is None:
+            return check
+
+        # skip extra allowed values
+        check_col = self.events[col]
+        if len(definitions[col]["value_range"]) == 3:
+            if definitions[col]["value_range"][2] is None:
+                check_col = check_col.dropna()
+            else:
+                check_col = check_col[check_col != definitions[col]["value_range"][2]]
+
+        # check value range for remaining values
+        if not (definitions[col]["value_range"][0] <= check_col).all():
+            check = False
+            warnings.warn(
```

Comment:
Same as above. I think for all checks you should decide whether the purpose is to check for a condition (column is "clean"), or evaluate problems of a column (provide a detailed report for definitions violations). As a user, I might be surprised if I wanted to "check definitions", and suddenly get a "False" plus two warnings. What's are they warning me for?

Reply:
My initial thought of combining the returned bool and the warnings was to limit the complexity in the event. This way, you can output detailed warnings (e.g. what value in which column failed the check) but also use the function to check the value ranges from the user side. 

However, I see the point of this being confusing. 

Therefore, I think best is to do a similar functionality as missing_essential_columns():
Returning the warnings exclusively in the post_init with a property/function self.incorrect_value_ranges() outputting the "dirty" columns.